HabboHotel/Groups/Forums/GroupForumThread.cs

```diff
@@ -0,0 +1,188 @@
+﻿using Dapper;
+using Plus.Communication.Packets.Outgoing;
+using Plus.Database;
+using Plus.HabboHotel.GameClients;
+using Plus.HabboHotel.Users;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Plus.HabboHotel.Groups.Forums
+{
+    public class GroupForumThread
+    {
+        public int Id;
+        public int UserId;
+        public string Caption;
+        public int Timestamp;
+
+        //Stats
+        public bool Pinned;
+        public bool Locked;
+        public int DeletedLevel;
+        public int DeleterUserId;
+        public int DeletedTimestamp;
+
+        private readonly IDatabase _database;
+
+        public GroupForum ParentForum;
+        public List<GroupForumThreadPost> Posts;
+
+        public List<GameClient> UsersOnThread;
+        public List<GroupForumThreadPostView> Views;
+
+
+        public GroupForumThread(GroupForum parent, IDatabase database, int id, int userid, int timestamp, string caption, bool pinned, bool locked, int deletedlevel, int deleterid)
```

Comment:
Personally not really excited to see services injected into a model class. I dont think this model is responsible to update itself in the database. I think whoever modifies the object should push the changes to the database via a dedicated services.